File: code/java/myAtoi.java
class Solution {
    public int myAtoi(String str) {
        str = str.trim();
        str = str.split(" ")[0];
        int INT_MAX = (int)(Math.pow(2,31) -1);
        int INT_MIN = (int)(-Math.pow(2,31));
        int start = -2;
        int flag = 0;
        int end = str.length();
        char c;
        for(int i=0;i<str.length();i++){
            c = str.charAt(i);
            if('0'<= c & c<='9'){
                if(flag==0){
                start = i;
                flag = 1;
                }

            }
            else
                if(flag == 1){
                    end = i;
                    break;
                }
            
        }
        //System.out.println(end);
        if(start == 1){
            if(str.charAt(0) == '+'){
                try{
                    return Integer.valueOf(str.substring(start,end));
                }
                catch(Exception e){
                    return INT_MAX;
                }
                    
            }
            if(str.charAt(0) == '-'){
                try{
                    return -Integer.valueOf(str.substring(start,end));
                }
                catch(Exception e){
                    return INT_MIN;
                }
            }
        }
        else if(start == 0){
                try{
                    return Integer.valueOf(str.substring(start,end));
                }
                catch(Exception e){
                    return INT_MAX;
                }           
        }
        System.out.println(start);
        return 0;
    
        


    }
}

File: code/python/myAtoi.py
class Solution:
    def myAtoi(self, str: str) -> int:
        lst = str.strip().split()
        if(len(lst) == 0):
            return 0
        flag = 0

        temp = lst[0]
        end = len(temp)

        for i in range(len(temp)):
            if flag == 0 and temp[i] >= "0" and temp[i] <= "9":
                flag = 1
                continue
            if flag == 1 and (temp[i] < "0" or temp[i] > "9"):
                end = i  
                flag = 2
                break

        try:
            num = int(temp[0:end])

            if num < 0:
                return max(num,-2147483648)
            elif num > 0:
                return min(num,2147483647)
            else:
                return 0
        except:
            return 0
